tenets/codelingo/jenkinsx/intformat/codelingo.yaml

```diff
@@ -0,0 +1,250 @@
+    query: |
+      import codelingo/ast/go
+
+      go.dir(depth = any):
+        go.file(depth = 0:2):
+          go.ident:
+            name as packageName
+            doesNotMatch(/test/, packageName)
+        @review.comment
+        go.file:
+          # The only top level ident in a go file is (should be) the package name
+          go.ident:
+            name == append(packageName, "_test")
+          filename as fname
+          doesNotMatch(/_integration_test.go/, fname)
+          # filename != append(packageName, "_integration_test.go")
+
+  - name: right-file-wrong-build-directive
+    doc: |
+      Integration tests should have the custom build directive `// +build integration`,
+      belong to a file called `.*_integration_test.go`, and a a package called `<packagename>_test`
+    flows:
+      codelingo/review:
+        comment: |
+          right-file-wrong-build-directive
+          Integration tests should have the custom build directive `// +build integration`,
+          belong to a file called `.*_integration_test.go`, and a a package called `<packagename>_test`
+    query: |
+      import codelingo/ast/go
+
+      go.dir(depth = any):
+        go.file(depth = 0:2):
+          go.ident:
+            name as packageName
+            doesNotMatch(/test/, packageName)
+        @review.comment
+        go.file:
+          filename as fname
+          regex(/integration_test.go/, fname)
+          # filename == append(packageName, "integration_test.go")
+          exclude:
+            go.comment_group:
+              go.comment:
+                start_offset == 0
+                text == "// +build integration"
+
+  - name: right-file-wrong-package
+    doc: |
+      Integration tests should have the custom build directive `// +build integration`,
+      belong to a file called `.*_integration_test.go`, and a a package called `<packagename>_test`
+    flows:
+      codelingo/review:
+        comment: |
+          right-file-wrong-package
+          Integration tests should have the custom build directive `// +build integration`,
+          belong to a file called `.*_integration_test.go`, and a a package called `<packagename>_test`
+    query: |
+      import codelingo/ast/go
+
+      go.dir(depth = any):
+        go.file(depth = 0:2):
+          go.ident:
+            name as packageName
+            doesNotMatch(/test/, packageName)
+        @review.comment
+        go.file:
+          filename as fname
+          regex(/integration_test.go/, fname)
+          # filename == append(packageName, "integration_test.go")
+          exclude:
+            # The only top level ident in a go file is (should be) the package name
+            go.ident:
+              name as testPackagename
+              regex(/_test/, testPackagename)
+              # name == append(packageName, "_test")
+
+## Better solutions:
+# Match files that have some characteristic of an integration test and are missing
```

Comment:
I don't understand this sentence? Files that have a characteristic are missing a characteristic?